### ISSUE TITLE
2.0 fails to run on Gnome 41

### ISSUE THREAD

**Issue by tcitworld** [SeaDve/Kooha#118]
I'm running Fedora 35 with Gnome 41.

When trying to record, the following issue happens:
```
Portal request failed: Unknown: Unknown
```
It outputs the following when run from terminal.
```
ERROR kooha::backend::screencast_portal > Error from screencast call: Portal(ZBus(Variant(Message("missing field `session_handle`"))))
```

The issue doesn't happen on 1.2.1

Possibly related to https://github.com/SeaDve/Kooha/pull/100 ?

**Comment by SeaDve** [comment]
hmm, i think it may be an issue within the library it uses to talk to screencast portal

**Comment by tcitworld** [comment]
Running with `RUST_LOG=kooha=debug` fives only the following:

```
INFO  kooha::application > Kooha (io.github.seadve.Kooha)
INFO  kooha::application > Version: 2.0.0 ()
INFO  kooha::application > Datadir: /app/share/kooha
DEBUG kooha::backend::recorder > is_show_pointer: false
DEBUG kooha::backend::recorder > is_selection_mode: true (false gives the same issue)
INFO  kooha::backend::screencast_portal > ScreenCastProxy created
ERROR kooha::backend::screencast_portal > Error from screencast call: Portal(ZBus(Variant(Message("missing field `session_handle`"))))
```

`xdg-desktop-portal` is at `1.10.1`, `xdg-desktop-portal-gtk` at 1.10.0 (which are quite recent, and both mention screencast changes).

**Comment by SeaDve** [comment]
I kinda suspect this commit https://github.com/flatpak/xdg-desktop-portal/commit/e4523531cfc478036fc4554e1d1f0582bb6bdf59

**Comment by xscd** [comment]
Yes, I had this too it also broke Firefox screen sharing. Its reverted now and everything works fine. If you are on Silverblue you won't get the update since its in Freeze try the testing branch of 35.

**Comment by SeaDve** [comment]
I'm still on 34, btw.

Anyway, I think we can mark this as out of scope for Kooha?

**Comment by xscd** [comment]
The issue owner has the latest updates. Maybe it could be something else?
```
xdg-desktop-portal-gtk-1.10.0-1.fc35.x86_64
xdg-desktop-portal-1.10.1-1.fc35.x86_64
xdg-desktop-portal-gnome-41.0-1.fc35.x86_64
```

Mine are like these and everything works. @tcitworld can you try sharing your screen on Firefox? https://mozilla.github.io/webrtc-landing/gum_test.html

**Comment by xscd** [comment]
@tcitworld make sure you are not on ```xdg-desktop-portal-gnome-41~alpha-1.fc35.x86_64```

**Comment by tcitworld** [comment]
I have the latest versions, but https://github.com/flatpak/xdg-desktop-portal/commit/e4523531cfc478036fc4554e1d1f0582bb6bdf59 does not seem to be in `xdg-desktop-portal` `1.10.1`, hence the issue.

Screensharing works in latest Chrome, but not in Firefox.

I'm closing this in favour of the bugzilla issue.

**Comment by tcitworld** [comment]
Seems to be fixed with `xdg-desktop-portal-gnome-41.1-1.fc35`, FYI.
